Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php

```diff
@@ -23,17 +24,25 @@
     protected $role = null;
 
     /**
-     * @param string $name
-     * @param string $role
-     * @param int    $weight
-     * @param string $parent
+     * @param string                   $name
+     * @param string                   $role
+     * @param int                      $weight
+     * @param string                   $parent
+     * @param array                    $datatableParameter
+     * @param TranslatorInterface|null $translator
      */
-    public function __construct($name, $role, $weight, $parent)
+    public function __construct($name, $role, $weight, $parent, array $datatableParameter = array(), $translator = null)
     {
         $this->name = $name;
         $this->role = $role;
         $this->weight = $weight;
         $this->parent = $parent;
+
+        if ($translator instanceof TranslatorInterface) {
+            $datatableParameter = $this->preFormatDatatableParameter($datatableParameter, $translator);
+        }
+        $this->datatableParameter = $datatableParameter;
+        $this->translator = $translator;
```

Comment:
why add translator in parameter of class if you don''t use it.
if you add translator in parameter of class, remove it of method preFormatDatatableParameter